source/common/http/http1/balsa_parser.cc

```diff
@@ -13,9 +13,25 @@
 namespace Http {
 namespace Http1 {
 
+namespace {
+
 using ::quiche::BalsaFrameEnums;
 using ::quiche::BalsaHeaders;
 
+bool isMethodValid(absl::string_view method) {
+  static constexpr absl::string_view kValidMethods[] = {
+      "ACL",       "BIND",    "CHECKOUT", "CONNECT", "COPY",       "DELETE",     "GET",
+      "HEAD",      "LINK",    "LOCK",     "MERGE",   "MKACTIVITY", "MKCALENDAR", "MKCOL",
+      "MOVE",      "MSEARCH", "NOTIFY",   "OPTIONS", "PATCH",      "POST",       "PROPFIND",
+      "PROPPATCH", "PURGE",   "PUT",      "REBIND",  "REPORT",     "SEARCH",     "SOURCE",
+      "SUBSCRIBE", "TRACE",   "UNBIND",   "UNLINK",  "UNLOCK",     "UNSUBSCRIBE"};
+
+  return std::binary_search(&kValidMethods[0], &kValidMethods[0] + ABSL_ARRAYSIZE(kValidMethods),
```

Comment:
This error is bogus.  sizeof() (expanded from ABSL_ARRAYSIZE) returns 34, the number of elements in the C-style array kValidMethods.  It would indeed be a mistake to add 34 * sizeof(string_view *) instead of 34 to a string_view*, so adding an expression including sizeof() to a pointer is often a mistake, but this is a false positive.

Reply:
Done.